sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/ResumeFromBackgroundTaskTest.java

```diff
@@ -0,0 +1,116 @@
+import com.microsoft.appcenter.utils.AsyncTaskUtils;
+import com.microsoft.appcenter.utils.storage.SharedPreferencesManager;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.rule.PowerMockRule;
+
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.isA;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+
+@PrepareForTest({
+        AsyncTaskUtils.class,
+        AppCenterLog.class,
+        Distribute.class,
+        ResumeFromBackgroundTask.class,
+        SharedPreferencesManager.class
+})
+public class ResumeFromBackgroundTaskTest {
+
+    /**
+     * Log tag for this service.
+     */
+    private static final String LOG_TAG = AppCenter.LOG_TAG + "Distribute";
+
+    @Rule
+    public PowerMockRule mRule = new PowerMockRule();
+
+    @Mock
+    private Context mContext;
+
+    @Mock
+    private Distribute mDistribute;
+
+    @Before
+    public void setUp() {
+        mockStatic(AsyncTaskUtils.class);
+        mockStatic(Distribute.class);
+        mockStatic(SharedPreferencesManager.class);
+        when(Distribute.getInstance()).thenReturn(mDistribute);
+    }
+
+    @Test
+    public void doInBackgroundInvalidId() {
+        when(SharedPreferencesManager.getLong(anyString(), anyLong())).thenReturn(-1L);
+
+        /* Start. */
+        startDoInBackground();
+        ResumeFromBackgroundTask task = AsyncTaskUtils.execute(LOG_TAG, new ResumeFromBackgroundTask(mContext, 1L));
+        task.doInBackground(null);
+
+        /* Verify. */
+        verify(mDistribute).startFromBackground(mContext);
+        verify(mDistribute, never()).resumeDownload();
+    }
+
+    @Test
+    public void doInBackgroundNotExpectedId() {
+        when(SharedPreferencesManager.getLong(anyString(), anyLong())).thenReturn(2L);
+
+        /* Start. */
+        startDoInBackground();
+        ResumeFromBackgroundTask task = AsyncTaskUtils.execute(LOG_TAG, new ResumeFromBackgroundTask(mContext, 4L));
+        task.doInBackground(null);
+
+        /* Verify. */
+        verify(mDistribute).startFromBackground(mContext);
+        verify(mDistribute, never()).resumeDownload();
+    }
+
+    @Test
+    public void doInBackgroundResumeDownloaded() {
+        long downloadedId = 2L;
+        when(SharedPreferencesManager.getLong(anyString(), anyLong())).thenReturn(downloadedId);
+
+        /* Start. */
+        startDoInBackground();
+        ResumeFromBackgroundTask task = AsyncTaskUtils.execute(LOG_TAG, new ResumeFromBackgroundTask(mContext, downloadedId));
+        task.doInBackground(null);
```

Comment:
I'm curious why we changed all varargs to an explicit array because now we have to call with null in tests everywhere.